int64 theta1
int64 theta2
int64 theta3
